### ISSUE TITLE
Request: Publish to npm

### ISSUE THREAD

**Issue by jimf** [simeydotme/jQuery-ui-Slider-Pips#59]
Would be handy to have this published to npm in addition to bower, since many projects are moving away from bower.

**Comment by simeydotme** [comment]
Thanks for the feedback, @jimf  :) -- I'll look in to it, but isn't it possible for NPM to specify a github repo as the dependency? :)

**Comment by jimf** [comment]
It is. However, you lose the version tracking, i.e., it'd be up to the user to manage the dependency manually, as tools like `npm outdated` etc. won't have insight into the github package (afaik).

**Comment by jamescoletti** [comment]
In the meantime, npm is complaining about the `main` property in `package.json` being an array. It expects a string pointing to the entry point of the module/script.

```
$ npm install git://github.com/simeydotme/jQuery-ui-Slider-Pips.git#v1.10.5
npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/Cellar/node/4.1.0/bin/node" "/usr/local/bin/npm" "install" "git://github.com/simeydotme/jQuery-ui-Slider-Pips.git#v1.10.5"
npm ERR! node v4.1.0
npm ERR! npm  v3.3.9

npm ERR! Path must be a string. Received [ 'dist/jquery-ui-slider-pips.css',
npm ERR!   'dist/jquery-ui-slider-pips.min.js' ]
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
...

**Comment by simeydotme** [comment]
any assistance on this would be appreciated :smile:  -- I have no experience with NPM packages

**Comment by jimf** [comment]
Once you have created an [npm account](https://www.npmjs.com/), it should just be a matter of running `npm publish` with your account credentials (you can create a ~/.npmrc to not specify these every time).

Also, I believe it's recommended, if not required, for the package.json name field to be in all lowercase.

**Comment by simeydotme** [comment]
thanks @jimf ... I understand that.
So I only had the package.json originally so Grunt could run... 

I also do not have any intention for this to be a exported anonymous module as it is an extension method for jqueryui... Am I right in thinking that NPM modules should always be self-executing functions, and not extensions for other libraries? -- This is the part I require some help/advice on :)

I do not want to wrap the extentions in an exported module, as that will make it inaccessible for non-npm/browserify users... or should I have anohter, new file, explicitly for browserify/npm which is wrapped as an exported module, and should it be classified as the "main" file?

Thanks for any insight.

**Comment by jimf** [comment]
Ah, gotcha. Yeah, I suppose it comes down to how much you want to put into it. FWIW, jQuery officially moved their [plugin repository](http://plugins.jquery.com/) to npm. The npm blog posted a couple of articles ([here](http://blog.npmjs.org/post/111385956875/welcome-jquery-developers) and [here](http://blog.npmjs.org/post/112712169830/making-your-jquery-plugin-work-better-with-npm)) on making the transition. But yeah, a true CommonJS setup would look a little different. You likely could always use a UMD wrapper though to keep the existing sort of plug-and-play support.

**Comment by simeydotme** [comment]
Excellent, @jimf -- thanks for your help.
I'll read through those links and try to come up with a solution! :smile:

And thanks for your patience.

Simon.

**Comment by jamescoletti** [comment]
Yep, that's the article. Thanks @jimf. The last few code snippets in the 2nd npm blog post describe how to handle both browser and CommonJS environments using a simple factory. Module exporting can be useful for things like unit testing or attaching the jQuery plugin manually for more specialized use cases. Thanks for looking into this.

**Comment by jimf** [comment]
Glad to help! :beers:

**Comment by ramytamer** [comment]
I still can't install it via npm.

**Comment by trailofdad** [comment]
+1 would love to have this in NPM

**Comment by simeydotme** [comment]
Hey, anyone who's interested in this being available on NPM, could you please review the changes on the 2nd of Oct, https://github.com/simeydotme/jQuery-ui-Slider-Pips/commits/2.0.0-alpha -- if anyone can test this on browserify/webpack I'd be greatly appreciative... if that works I'll release it as 2.0.0 and push to NPM :)

**Comment by simeydotme** [comment]
- [x] push `2.0.0-alpha` with AMD wrapper
- [x] fix up `package.json` file
- [ ] test `2.0.0-alpha` being includable via NPM
- [ ] test inclusion via `require()` with *webpack* and *browserify*

**Comment by simeydotme** [comment]
----
*edited for milestone tracking*

- [x] push `2.0.0-alpha` with AMD wrapper
- [x] fix up `package.json` file
- [ ] test `2.0.0-alpha` being includable via NPM
- [ ] test inclusion via `require()` with *webpack* and *browserify*
- [ ] release 2.0.0
- [ ] publish on NPM

**Comment by Deckluhm** [comment]
> Also, I believe it's recommended, if not required, for the package.json name field to be in all lowercase.

https://docs.npmjs.com/getting-started/using-a-package.json#requirements

**Comment by khandelwalriya** [comment]
How can we get values of the slider to perform some operation based on slider value?
